controllers/article-controller.js

```diff
@@ -11,9 +12,32 @@ exports.getArticleById = (req, res, next) => {
 };
 
 exports.getArticles = ((req, res, next) => {
-    return getArticlesQuery(next)
+    const topic = req.body.topic;
+    let promises = [];
+
+    if (topic && typeof topic !== 'string') {
+        res.status(400).send({msg: 'Bad request'})
+    }
+    else if (topic  && topic !== '') {
+        // filtered by topic
+        promises.push(getTopic(topic, next));
+        promises.push(getArticlesQuery(next, topic));
+    }
+    else {
+        // not filtered by topic
+        promises.push(getArticlesQuery(next))
+    }
+
+    Promise.all(promises)
```

Comment:
It's quite inefficient to make two queries for getArticlesQuery(), it'd be ideal to just make one, based on whether the user passes in a topic or not

Reply:
Done :)